File: 프로그래머스/lv2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    const possibleResults = {}
    const result = []
    record.forEach((r) => {
        const [status, id, name] = r.split(" ")
        
        if (status === "Enter" || status === "Change"){
            possibleResults[id] = name
        }
    })
    
    record.forEach((r) => {
        const [status, id, name] = r.split(" ")
        if (status === "Enter") {
            result.push(possibleResults[id] + "님이 들어왔습니다.")
        } else if (status === "Leave") {
            result.push(possibleResults[id] + "님이 나갔습니다.")
        }
    })
    return result
}

File: 프로그래머스/lv2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    result = []
    name = {}  # 딕셔너리
    for i in record:
        check1 = i.split()
        if check1[0] == 'Enter':
            name[check1[1]] = check1[2]
        elif check1[0] == 'Change':
            name[check1[1]] = check1[2]
    #  name = {'uid1234': 'Prodo', 'uid4567': 'Ryan'}
    
    for i in record:
        check2 = i.split()
        if check2[0] == 'Enter':
            result.append(f'{name[check2[1]]}님이 들어왔습니다.')
        elif check2[0] == 'Leave':
            result.append(f'{name[check2[1]]}님이 나갔습니다.')
    return result